### PR TITLE
Review of DM-3387 ("Make use of good pixel count when building CoaddPsfs")

### PR DESCRIPTION
**Pull request by jdswinbank**

### CODE REVIEW

File: src/CoaddPsf.cc [hunk 1]
```diff
@@ -152,9 +152,18 @@ CoaddPsf::CoaddPsf(
 {
     afw::table::SchemaMapper mapper(catalog.getSchema());
     mapper.addMinimalSchema(afw::table::ExposureTable::makeMinimalSchema(), true);
+
+    // copy the field "goodpix", if available, since computeAveragePosition will use it, if available
+    try {
+        afw::table::Key<int> goodPixKey = catalog.getSchema()["goodpix"]; // auto does not work
+        mapper.addMapping(goodPixKey, true);
+    } catch (pex::exceptions::NotFoundError &) {}
```
Comment (jdswinbank) [inline src/CoaddPsf.cc]: A pretty trivial point, but: you've used "if available" twice one one line above, plus a further twice in the commit message, plus one instance of "if it was available". None of them are actually wrong, but all that repetition makes things pretty hard to read! Could you spend a few seconds rephrasing for clarity?

File: tests/CoaddPsf.py [hunk 2]
```diff
@@ -28,27 +28,16 @@
    python CoaddPsf.py
 """
 
-import os, sys
-from math import *
-import numpy
```
Comment (jdswinbank) [inline tests/CoaddPsf.py]: Thank you for all the cleaning up you've done in this file!

File: tests/CoaddPsf.py [hunk 3]
```diff
@@ -534,6 +510,48 @@ def testValidPolygonPsf(self):
...
+        for i, (xOff, yOff, numGoodPix) in enumerate((
+            (30.0, -20.0, 25),
+            (32.0, -21.0, 10),
+            (28.0, -19.0, 30),
+        )):
+            xwsum -= xOff * numGoodPix
+            ywsum -= yOff * numGoodPix
+            wsum += numGoodPix
+            record = mycatalog.getTable().makeRecord()
+            record.setPsf(measAlg.DoubleGaussianPsf(25, 25, 10, 1.00, 0.0))
```
Comment (jdswinbank) [inline tests/CoaddPsf.py]: When I see numbers hard-coded like this in a test case, I often find it helpful if there's an accompanying comment explaining why those particular numbers were chosen -- are they completely arbitrary placeholders, or do they have some significance regarding the test conditions? In this case, it's pretty clear what's happening so I don't really object, but maybe it's worth considering.

File: tests/CoaddPsf.py [hunk 4]
```diff
@@ -534,6 +510,48 @@ def testValidPolygonPsf(self):
...
+        crval1, crval2 = 0.0, 0.0
+        bboxSize = afwGeom.Extent2I(2000, 2000)
+        crpix = afwGeom.PointD(bboxSize/2.0)
+        crval = afwCoord.Coord(afwGeom.Point2D(crval1, crval2))
+        wcsref = afwImage.makeWcs(crval, crpix, cd11, cd12, cd21, cd22)
+
+        schema = afwTable.ExposureTable.makeMinimalSchema()
+        schema.addField("weight", type="D", doc="Coadd weight")
+        schema.addField("goodpix", type="I", doc="Number of good pixels")
+        mycatalog = afwTable.ExposureCatalog(schema)
```
Comment (jdswinbank) [inline tests/CoaddPsf.py]: Being especially picky, I might question whether some of the variable names here match the coding standard. Should it be `myCatalog`, `wSum`, etc?

Comment (r-owen) [inline tests/CoaddPsf.py]: I agree, but they are pre-existing names and things like crpix are a borderline case. I'd prefer to leave it alone.

File: src/CoaddPsf.cc (hunk 1 above)
Comment (r-owen) [inline src/CoaddPsf.cc]: OK

File: tests/CoaddPsf.py (hunk 3 above)
Comment (r-owen) [inline tests/CoaddPsf.py]: OK

**Pull request merged**